extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -465,27 +494,61 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.updateInViewport(dev().assertElement(this.container_), true);
     this.scheduleLayout(dev().assertElement(this.container_));
 
-    this.currentElementId_ = element.lightboxItemId;
-    // Hack to access private property. Better than not getting
-    // type checking to work.
-    /**@type {?}*/ (this.carousel_).implementation_.showSlideWhenReady(
-        this.currentElementId_);
-
     this.win.document.documentElement.addEventListener(
         'keydown', this.boundHandleKeyboardEvents_);
 
-    return Promise.resolve();
+    return this.resources_.requireLayout(this.carousel_)
+        .then(() => {
+          this.currentElemId_ = element.lightboxItemId;
+          this.carousel_.implementation_
+              .showSlideWhenReady(this.currentElemId_);
+          this.resizeImageViewerDimensions_();
+          this.updateDescriptionBox_();
+        });
+  }
+
+  /**
+   * Resizes the ImageViewer of the current slide so that the image
+   * is centered in the page. Used in open or on slide change. Also
+   * registers a onResize handler to resize the ImageViewer whenever
+   * the screen size changes.
+   * @private
+   */
+  resizeImageViewerDimensions_() {
+    const imgViewer = this.elementsMetadata_[this.currentElemId_].imageViewer;
+    imgViewer.measure();
+
+    this.unlistenViewport_ = this.getViewport().onResize(() => {
+      // In IOS 10.3, the measured size of an element is incorrect if the
+      // element size depends on window size directly and the measurement
+      // happens in window.resize event. Adding a timeout for correct
+      // measurement. See https://github.com/ampproject/amphtml/issues/8479
+      if (startsWith(
+          Services.platformFor(this.win).getIosVersionString(), '10.3')) {
+        Services.timerFor(this.win).delay(() => {
+          imgViewer.measure();
+        }, 500);
+      } else {
+        imgViewer.measure();
+      }
+    });
   }
 
   /**
    * Closes the lightbox-viewer
    * @private
    */
+
```

Comment:
And I know you only copied this code, but I'm not sure about the return type of this function. Is the promise returned ever used? If not, it can probably just return nothing.